TSOClient/tso.files/FAR1/FAR1Archive.cs

```diff
@@ -65,18 +65,26 @@ public FAR1Archive(string Path, bool v1b)
             m_Reader.BaseStream.Seek(m_ManifestOffset, SeekOrigin.Begin);
 
             m_NumFiles = m_Reader.ReadUInt32();
-            
+
+
             for (int i = 0; i < m_NumFiles; i++)
             {
-                FarEntry Entry = new FarEntry();
-                Entry.DataLength = m_Reader.ReadInt32();
-                Entry.DataLength2 = m_Reader.ReadInt32();
-                Entry.DataOffset = m_Reader.ReadInt32();
-                Entry.FilenameLength = (v1b)?m_Reader.ReadInt16():(short)m_Reader.ReadInt32();
-                Entry.Filename = Encoding.ASCII.GetString(m_Reader.ReadBytes(Entry.FilenameLength));
-
-                m_Entries.Add(Entry);
-            }
+                try
+                {
+                    FarEntry Entry = new FarEntry();
+                    Entry.DataLength = m_Reader.ReadInt32();
+                    Entry.DataLength2 = m_Reader.ReadInt32();
+                    Entry.DataOffset = m_Reader.ReadInt32();
+                    Entry.FilenameLength = (v1b) ? m_Reader.ReadInt16() : (short)m_Reader.ReadInt32();
+                    Entry.Filename = Encoding.ASCII.GetString(m_Reader.ReadBytes(Entry.FilenameLength));
+
+                    m_Entries.Add(Entry);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"FAR1 Archive at: {m_Path} loading entry number: {i} threw an exception.", e);
+                }
```

Comment:
I'm not sure anything could throw in here unless the files were corrupt, and it's outside the scope of this PR.